API/common/exception.py

```diff
@@ -1,25 +1,72 @@
-class ParameterException(Exception):
-    def __init__(self, msg: str) -> None:
+from common.util import generate_response
+
+
+class RequestException(Exception):
+    ERROR_CODE = 400
+
+    def __init__(self, msg: str):
         super().__init__()
         self.msg = msg
 
+    def to_response_body(self) -> str:
+        return generate_response([], self.ERROR_CODE, self.msg)
+
+
+class ParameterException(RequestException):
+    ERROR_CODE = 401
 
-class CopilotException(Exception):
     def __init__(self, msg: str) -> None:
+        super().__init__(msg)
+
+
+class QuestionOutOfScopeException(RequestException):
+    ERROR_CODE = 402
+
+    def __init__(self, msg: str) -> None:
+        super().__init__(msg)
+
+
+class ServiceException(Exception):
+    ERROR_CODE = 502
```

Comment:
Could we start the error code with 500?

Reply:
`500` is used for other exceptions caught by service which are not defined in `RequestException` nor `ServiceException`. `501` is remained for the deleted `CopilotException`.

Reply:
OK, how about starting from `501`? Because `501` has not been used yet, skipping the design of `501` would be a bit strange. Since `CopilotException` may not be a service issue, starting with `5xxx` may not necessarily be a suitable choice

Reply:
Will do